fix(search indexes): support for async search index settings

easen-amp opened the pull request:
Depends upon on https://github.com/amplience/dc-management-sdk-js/pull/126

Updated to work using the recent algolia search api change.

rs-amp commented:
Trying to import/update a search index with replicas seems to give me this error:
```
TypeError: resource.toJSON is not a function TypeError: resource.toJSON is not a function
    at SearchIndex.<anonymous> (/Users/rsimpson/Documents/amplience/dc-cli/node_modules/dc-management-sdk-js/build/main/lib/model/SearchIndex.js:141:77)
    at step (/Users/rsimpson/Documents/amplience/dc-cli/node_modules/dc-management-sdk-js/build/main/lib/model/SearchIndex.js:46:23)
    at Object.next (/Users/rsimpson/Documents/amplience/dc-cli/node_modules/dc-management-sdk-js/build/main/lib/model/SearchIndex.js:27:53)
    at fulfilled (/Users/rsimpson/Documents/amplience/dc-cli/node_modules/dc-management-sdk-js/build/main/lib/model/SearchIndex.js:18:58)
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
```
It happens on this line:
```
  // Update the search index settings.
  await index.related.settings.update(enrichedIndex.settings, false, {
    waitUntilApplied: hasReplicas ? ['replicas'] : false
  });
```

It looks like it's expecting a SearchIndexSettings instance for `enrichedIndex.settings`, but it's actually just a JSON object. Wrapping it with `new SearchIndexSettings(...)` seems to work.

easen-amp commented:
@rs-amp It seems the `EnrichedSearchIndex` doesn't correctly create various types (settings, etc.) when the data is loaded from disk.

I've updated the test to ensure the settings is an instance of `SearchIndexSettings`. Then I've corrected the `EnrichedSearchIndex` class convert the `settings`, `keys`, `replicas` & `assignedContentTypes` into their correct type, as well updating the `toJSON()` to invoke each property's `toJSON()` method. 

[See my last commit for the diff of all my changes.](https://github.com/amplience/dc-cli/pull/165/commits/715936122fe1efd2642767f67ea2b92fc4f3f3d1)

rs-amp reviewed (approved):
LGTM. 🥳
